src/config.c

```diff
@@ -1,125 +1,351 @@
 #include "config.h"
+#include "err.h"
 #include "rmutil/util.h"
 #include "rmutil/strings.h"
 #include <string.h>
 #include <stdlib.h>
+#include <limits.h>
 
-RSConfig RSGlobalConfig = RS_DEFAULT_CONFIG;
+#define RETURN_ERROR(s) return REDISMODULE_ERR;
 
-int ReadConfig(RedisModuleString **argv, int argc, const char **err) {
-  *err = NULL;
-  /* Read the extension we want to load */
-  if (argc > 1 && RMUtil_ArgIndex("EXTLOAD", argv, argc) >= 0) {
-    RMUtil_ParseArgsAfter("EXTLOAD", argv, argc, "c", &RSGlobalConfig.extLoad);
+static int readLongLong(RedisModuleString **argv, size_t argc, size_t *offset, long long *out) {
+  if (*offset == argc) {
```

Comment:
think its better to do '>='

Reply:
offset should never exceed argc. If it does, I'd like it to crash rather than hide a bug somewhere

Reply:
So lets put an assert on this case